File: java/java-maven/src/main/java/hello/HelloController.java
package hello;

import java.net.InetAddress;
import java.net.UnknownHostException;

import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class HelloController {

    @RequestMapping("/")
    public String index() throws UnknownHostException {
        var hostname = InetAddress.getLocalHost().getHostName();
        return String.format("[Java-Maven] Welcome to %s!", hostname);
    }
    
}

File: scala/src/main/scala/hello/HelloController.scala
package hello

import java.net.{InetAddress,UnknownHostException}
import org.springframework.web.bind.annotation.{RequestMapping,RestController}

@RestController
class HelloController {

    @RequestMapping(Array("/"))
    @throws(classOf[UnknownHostException])
    def index = {
        val hostname = InetAddress.getLocalHost().getHostName()
        s"[Scala] Welcome to ${hostname}"
    }
    
}
